frontend/pages/hosts/details/cards/Software/Software.tsx

```diff
@@ -250,8 +250,8 @@ const SoftwareCard = ({
         <Spinner />
       ) : (
         <>
-          {(isError || !data) && <DataError />}
-          {!isError && data && (
+          {isError && <DataError />}
```

Comment:
sanity check: no need to check for a null/undefined `data` anymore?

Reply:
Yeah, that was part of the problem. We were displaying an error state instead of empty state because the API call was disabled (thus undefined data). The way these components are composed is not ideal. Too many places with shared responsibility for table states.